packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -235,12 +236,29 @@ public function getNativeType(Expr $expr): Type
             }
         }
 
-        /**
-         * ArrayDimFetch got type by its Scope->getType(), otherwise, it got MixedType
-         */
-        $type = $expr instanceof ArrayDimFetch
-            ? $scope->getType($expr)
-            : $scope->getNativeType($expr);
+        $type = $scope->getNativeType($expr);
+        if ($expr instanceof ArrayDimFetch) {
+            /**
+             * Allow pull type from
+             *
+             *      - native function
+             *      - always defined by assignment
+             *
+             * eg:
+             *
+             *  $parts = parse_url($url);
+             *  if (!empty($parts['host'])) { }
+             *
+             * or
+             *
+             *  $parts = ['host' => 'foo'];
+             *  if (!empty($parts['host'])) { }
+             */
+            $variableType = $scope->getNativeType($expr->var);
+            if (! $variableType instanceof ArrayType || ! $variableType->getItemType() instanceof MixedType) {
```

Comment:
it seems mixed check need to be added here as well, I will add test for pull from `AddMethodCallBasedStrictParamTypeRector`

Reply:
https://github.com/rectorphp/rector-src/pull/5058